### ISSUE TITLE
Arguments for ParamSpec "P@__call__" are missing

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5196]
### Discussed in https://github.com/microsoft/pyright/discussions/5195

<div type='discussions-op-text'>

<sup>Originally posted by **PerchunPak** May 29, 2023</sup>
After https://github.com/microsoft/pyright/commit/d91130c18948bc8b52583a1d35790fc9040da6a0, we now get the error `Arguments for ParamSpec "P@__call__" are missing` for this code:

```py
from typing import TypeVar, Callable
from typing_extensions import ParamSpec, Protocol

P = ParamSpec("P")
R = TypeVar("R")

class Decorator(Protocol):
    def __call__(self, __x: Callable[P, R]) -> Callable[P, R]:
        ...

def func() -> Decorator:
    ...

func()(lambda: None)()  # error: Arguments for ParamSpec "P@__call__" are missing
```

The question is: what is wrong here? It works well with 1.1.309.</div>

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.311, which I just published. It will also be included in a future release of pylance.
